\begin{longtable}{@{}p{3.5cm}@{}p{\dimexpr\textwidth-1cm\relax}@{}}
\nomen{AC}{Alternating current}
\nomen{AGC}{Automatic gain control}
\nomen{API}{Application programming interface}
\nomen{a.u.}{Arbitrary unit}
\nomen{BCA}{Bicinchoninic acid protein assay}
\nomen{BLAST}{Basic local alignment search tool}
\nomen{CAD}{Collision-activated dissociation}
\nomen{CEO}{Calculated elution order}
\nomen{CI}{Chemical ionization}
\nomen{CID}{Collision-induced dissociation}
\nomen{COMPASS}{Coon OMSSA Proteomic Analysis Software Suite}
\nomen{C\#}{C sharp, a programming language}
\nomen{CSMSL}{C\# Mass Spectrometry Library}
\nomen{Da}{Dalton, the atomic mass unit}
\nomen{DC}{Direct current}
\nomen{DDA}{Data-dependent acquisition}
\nomen{DIA}{Data-independent acquisition}
\nomen{DNA}{Deoxyribonucleic acid}
\nomen{DT}{Decision tree}
\nomen{DTT}{Dithiothreitol}
\nomen{ECD}{Electron-capture dissociation}
\nomen{EO}{Elution order}
\nomen{EOA}{Elution order algorithm}
\nomen{ESI}{Electrospray ionization}
\nomen{ETD}{Electron-transfer dissociation}
\nomen{E-value}{Expectation value}
\nomen{FAB}{Fast-atom bombardment}
\nomen{FASTA}{A format for storing protein sequences}
\nomen{FDR}{False discovery rate}
\nomen{FT}{Fourier transform}
\nomen{FT-ICR}{Fourier transform ion cyclotron resonance}
\nomen{GC}{Gas chromatography}
\nomen{GUI}{Graphical user interface}
\nomen{HCD}{Higher-energy collisional dissociation}
\nomen{HPLC}{High-performance liquid chromatography}
\nomen{HTCondor}{High throughput condor}
\nomen{Hz}{Hertz, inverse seconds}
\nomen{IDA}{Intelligent data acquisition}
\nomen{INC}{Inclusion list}
\nomen{IP}{Intellectual property}
\nomen{\inseq{}}{Instant sequencing algorithm}
\nomen{IT}{Ion trap}
\nomen{ITCL}{Ion trap control language}
\nomen{LC}{Liquid chromatography}
\nomen{\emph{m}}{Mass}
\nomen{MALDI}{Matrix-assisted laser desorption-ionization}
\nomen{min}{Minute}
\nomen{MS}{Mass spectrometry}
\nomen{MS$^1$}{Survey mass analysis}
\nomen{MS$^n$}{Tandem mass spectrometry}
\nomen{MS/MS}{Tandem mass spectrometry}
\nomen{NCE}{Normalized collision energy}
\nomen{\mz{}}{Mass-to-charge ratio}
\nomen{nLC}{Nanoflow liquid chromatography}
\nomen{OMSSA}{Open Mass Spectrometry Search Algorithm}
\nomen{ppm}{Part per million}
\nomen{PRM}{Parallel reaction monitoring}
\nomen{PSM}{Peptide-spectrum match}
\nomen{PTM}{Post-translational modification}
\nomen{QM}{QuantMode}
\nomen{RP}{Reverse phase}
\nomen{RT}{Retention time}
\nomen{s}{Second}
\nomen{SDF}{Site-determining fragment}
\nomen{SILAC}{Stable isotope labeling by amino acids in cell culture}
\nomen{SIM}{Single ion monitoring}
\nomen{S/N}{Signal-to-noise ratio}
\nomen{SRM}{Selected reaction monitoring}
\nomen{SCX}{Strong-cation exchange}
\nomen{S/N}{Signal-to-noise ratio}
\nomen{Th}{Thomson, the unit of the mass-to-charge ratio}
\nomen{TIC}{Total-ion chromatogram}
\nomen{TMT}{Tandem mass tag}
\nomen{XIC}{Extracted-ion chromatogram}
\nomen{\emph{z}}{Charge}
\end{longtable}